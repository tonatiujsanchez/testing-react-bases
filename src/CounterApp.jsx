import { useState } from "react"

import PropTypes from "prop-types"


export const CounterApp = ({ value }) => {
    
    const [ counter, setCounter ] = useState( value )

    const handleAdd = ( worth ) => {
        setCounter( counter + worth )
    }



    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            
            <button
                onClick={ ()=> handleAdd( -1 )} 
                className="btn"
                style={{
                    marginRight: '15px'
                }}
            >
                -1
            </button>
            <button
                onClick={ ()=> handleAdd( 1 )} 
                className="btn"
                style={{
                    marginRight: '15px'
                }}
            >
                +1
            </button>

            <button
                aria-label='btn-reset'
                onClick={ ()=> setCounter(value) } 
                className="btn"
            >
                Reset
            </button>
        </>
    )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
