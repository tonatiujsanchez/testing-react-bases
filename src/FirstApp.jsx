import PropTypes from 'prop-types'



export const FirstApp = ({ title, subtitle, name }) => {
    
    return (
        <div>
            <h1 data-testid="test-title" style={{ margin: 0 }}> { title } </h1>
            <h2 style={{ margin: 0 }}><strong>{ subtitle }</strong></h2>
            <h2 style={{ margin: 0 }}><strong>{ subtitle }</strong></h2>
            <p style={{ margin: 0 }}>{ name }</p>
        </div>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name: 'Tonatiuj Sácnhez',
    subtitle: 'Sin subtítulo',
    // title: 'Sin título',
}
