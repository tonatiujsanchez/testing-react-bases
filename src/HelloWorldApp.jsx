

export const HelloWorldApp = () => {
    return (
        <h1>HelloWorld App!!</h1>
    )
}
