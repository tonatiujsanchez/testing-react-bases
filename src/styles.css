


html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

.btn {
    padding: 5px 10px;
    font-size: 1rem;
    font-weight: bold;
    cursor: pointer;
}