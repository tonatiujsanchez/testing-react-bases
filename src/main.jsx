import React from 'react'
import ReactDOM from 'react-dom/client'


// import { HelloWorldApp } from './HelloWorldApp'
// import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'

import './styles.css'


ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <FirstApp title="Hi! i'm Ton.dev" /> */}
        <CounterApp value={ 100 } />
    </React.StrictMode>
)
