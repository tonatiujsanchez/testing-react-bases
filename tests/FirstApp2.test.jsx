import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Pruebas en el coponente FirstApp', () => {

    const title="Hola, soy Goku"
    const subtitle="I'm a subtitle"


    test('Debe de hacer match con el snapshot', () => {

        const { container } = render( <FirstApp title={ title } /> )
        expect( container ).toMatchSnapshot()

    })

    test('Debe de mostrar el mensahe "Hola, soy Goku"', () => {

        render( <FirstApp title={ title } /> )
        
        expect( screen.getByText(title) ).toBeTruthy()
        // expect( screen.getByText(title) ).not.toBeTruthy() //Para negar
    })


    test('Debe de mostar el titulo en un h1', () => {

        render( <FirstApp title={ title } /> )
        expect( screen.getByRole('heading',{ level: 1 }).innerHTML ).toContain( title )
    })

    

    test('Debe de mostrar el subtitulo enviado por props', () => {

        render( 
            <FirstApp 
                title={ title }
                subtitle= { subtitle }
            /> 
        )

        expect( screen.getAllByText(subtitle).length ).toBe(2)

    })


})