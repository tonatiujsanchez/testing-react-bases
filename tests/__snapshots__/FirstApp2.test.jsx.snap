// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el coponente FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
      style="margin: 0px;"
    >
       
      Hola, soy Goku
       
    </h1>
    <h2
      style="margin: 0px;"
    >
      <strong>
        Sin subtítulo
      </strong>
    </h2>
    <h2
      style="margin: 0px;"
    >
      <strong>
        Sin subtítulo
      </strong>
    </h2>
    <p
      style="margin: 0px;"
    >
      Tonatiuj Sácnhez
    </p>
  </div>
</div>
`;
